meetings/2018-01-17/minutes.md

```diff
@@ -0,0 +1,417 @@
+talked about "@rules", and demonstrated how fast Pants is, citing some sub-50ms
+compilation times. Pants also offers dependency inference from source-code
+imports.
+
+He emphasized that he thought a "thin" LSP model would be the best approach,
+where a build tool proxies the thin LSP protocol to a thin LSP server to invoke
+a presentation compiler.
+
+### Bazel - P. Oscar Boykin, Stripe
+
+Bazel is a build tool developed by Google focussing on fast, pure, reproducible
+and correct builds, taking heavy advantage of caching. It's been used at Stripe
+for two years, and supports multiple different languages. Bazel uses external
+  source dependencies.
+
+The build language is a restricted subset to ensure reproducibility. Google
+have a team of fifty people working on Bazel.
+
+### Ensime LSP server - Vladimir Polushin & Rory Graves
+
+Vladimir had a bad connection during the call, but after the meeting he
+provided the following notes on the Ensime LSP server.
+
+Ensime supports IDE features in non-IDE editors, it uses a built index and the
+presentation compiler like LSP/Scalameta. Ensime supports LSP out-of-the-box
+now, but not all clients have been migrated to it. The LSP implementation is a
+port of Iulien Dragos's original glue layer implementing all of the same
+features. It is purely focused on IDE support (completion, reverse lookup,
+etc), and does not manage building features.
+
+It was funded by the sponsorship program which also supported a shared LSP
+client for VSCode/Atom, though this funding has basically run out. Ensime has
+support for exporting build info from sbt/Gradle/Maven.
+
+The indexer is the key component for many of the LSP semantic calls, and this
+should move to using SemanticDB as an interchange format. To provide all of the
+IDE features you need more than just the protocol layer, much of which would be
+shared.
+
+Rory added, during the call, that Ensime creates an index, much like the
+SemanticDB index, but predating SemanticDB. Ensime could support all clients
+once LSP support is available.
+
+### IntelliJ IDE - Nikolay Tropin, JetBrains
+
+Nikolay is the team leader for the Scala plugin for IntelliJ, which provides
+support for Scala in IntelliJ. The IntelliJ plugin does not depend on Scalac,
+instead using its own implementation of the Scala typechecker for providing
+presentation support in the IntelliJ editor. IntelliJ also offers Java support.
+
+JetBrains are very interested in LSP and BSP, though they're not sure whether
+they can use it in IntelliJ, which currently relies on their own
+implementation. He noted that Bloop may be particularly useful for IntelliJ,
+though is uncertain about whether it could easily replace IntelliJ's own
+typechecker.
+
+### ScalaIDE/Eclipse - Iulian Dragos, Triplequotes
+
+Iulian has been working on the LSP implementation for VSCode, based on Ensime.
+The implementation delegates the protocol as an in-process Ensime instance,
```

Comment:
Might be a small typo here, shouldn't it be 'to an in-process' instead of 'as an ..'